src/storage/src/hummock/sstable/forward_sstable_iterator.rs

```diff
@@ -52,14 +53,43 @@ pub struct SstableIterator {
     sstable_store: SstableStoreRef,
     stats: StoreLocalStatistic,
     options: Arc<SstableIteratorReadOptions>,
+    // is_current_block_valid: bool,
+    // // The key range of the iterator
+    // key_range: KeyRange,
+
+    // // used for checking if the block is valid
+    // // it's a light weight check only when the table id changes
+    // last_table_id: StateTableId,
+    read_block_meta_range: (usize, usize),
 }
 
 impl SstableIterator {
     pub fn new(
         sstable: TableHolder,
         sstable_store: SstableStoreRef,
         options: Arc<SstableIteratorReadOptions>,
+        sstable_info_ref: &SstableInfo,
     ) -> Self {
+        let mut start_idx = 0;
+        let mut end_idx = sstable.meta.block_metas.len() - 1;
+        let read_table_id_range = (
+            *sstable_info_ref.table_ids.first().unwrap(),
+            *sstable_info_ref.table_ids.last().unwrap(),
+        );
+        let block_meta_count = sstable.meta.block_metas.len();
+
+        while start_idx < block_meta_count
+            && sstable.meta.block_metas[start_idx].table_id().table_id() < read_table_id_range.0
+        {
+            start_idx += 1;
+        }
+
+        while end_idx > start_idx
+            && sstable.meta.block_metas[end_idx].table_id().table_id() > read_table_id_range.1
+        {
+            end_idx -= 1;
+        }
+
```

Comment:
We assume that the table id read must exist in the sstable, otherwise it is a fatal error.

let me add some assertions to provide the strongest possible assurance.